snowshu/adapters/target_adapters/base_target_adapter.py

```diff
@@ -135,9 +137,21 @@ def initialize_replica(self, source_adapter_name: str, override_image: str = Non
             override_image: the name of incremental image to initialize,
                 if specified will override default image
         """
-        if override_image:
-            self.__class__.DOCKER_IMAGE = override_image
         try:
+            if override_image:
+                shdocker = SnowShuDocker()
+                images = shdocker.client.images.list(name=override_image)
+                if len(images) > 0:
```

Comment:
Since we've got the generic `except` block, we can probably skip this check and log the found images before proceeding

If we leave this `if` statement in, then `else`  block would need to properly raise an error